docs/concepts/internationalization.md

```diff
@@ -0,0 +1,72 @@
+
+Localization will happen in the game's locale. In a Minecraft client the locale is specified by the language settings. On a dedicated server, the only supported locale is en_US. A list of available locales can be found on the [Minecraft Wiki](https://minecraft.gamepedia.com/Language#Available_languages).
+
+Language files
+--------------
+
+Language files are located by `assets/domain/lang/[locale].lang` (e.g. the US English translation for `examplemod` would be `assets/examplemod/lang/en_us.lang`). Resource pack format 3 requires the locale name to be lowercased. The file format is simply lines of key-value pairs separated by `=`. Lines starting with `#` are ignored. Lines without a separator are ignored. The file must be encoded in UTF-8.
+
+```properties
+# items
+item.examplemod.example_item.name=Example Item Name
+
+# blocks
+tile.examplemod.example_block.name=Example Block Name
+
+# commands
+commands.examplemod.examplecommand.usage=/example <value>
+```
+
+Including the comment `#PARSE_ESCAPES` anywhere in the file will enable the slightly different and more complex Java properties file format. This format is required to support multiline values.
+
+!!! note
+    Enabling `#PARSE_ESCAPES` enables changes various aspects of the parsing. Among other things, the `:` character will be treated as a key-value separator. It must either be excluded from translation keys or escaped using `\:`.
+
+Unlocalized names
+-----------------
+
+Blocks, Items and other Minecraft constructs have unlocalized names. Unlocalized names are simply translation keys that are used when displaying their names.
+
+Unlike registry names, unlocalized names are not namespaced. It is therefore conventional to have unlocalized names prefixed with your modid (e.g. `examplemod.example_item`) to avoid conflicts. Otherwise, in the event of a conflict, the localization of one item will override the other.
+
+!!! note
+    The only purpose of unlocalized names is internationalization. Do not use them for logic. Use registry names instead.
+
+    A common pattern is using `getUnlocalizedName().substring(5)` to assign registry names. This is fragile and uses the unlocalized name for logic, which is considered bad practice. It could instead be reversed so that the registry name is set first, then the unlocalized name is set according to the registry name `MODID + "." + getRegistryName().getResourcePath()`.
+
+Localization methods
+--------------------
+
+!!! warning
+    A common issue is having the server localize for clients. The server can only localize in its own locale, which does not necessarily match the locale of connected clients.
+    
+    To respect the language settings of clients, the server should have clients localize text in their own locale using TextComponentTranslation or other methods preserving the language neutral translation keys.
```

Comment:
`TextComponentTranslation` should be in code tags

Reply:
Yup